source/page/data.ts

```diff
@@ -5,105 +5,111 @@ import banner_2018_React from './2018-React/React-ChengDu.jpg';
+        },
+        {
+            title: t('coscon20_chengdu_branch_venue'),
+            date: '2020-10-24',
+            URL: 'https://fcc-cd.dev/activity/conference/coscon-2020-chengdu/',
+            banner: 'https://ows.blob.core.chinacloudapi.cn/files/assets/COS_Con_20_abaa9e2a66.png'
+        },
+        {
+            title: t('chengdu_web_full_stack_conference_2019'),
+            date: '2019-11-16',
+            URL: '2019/',
+            banner: banner_2019
+        },
+        {
+            title: t('chengdu_web_frontend_conference_2018'),
+            date: '2018-11-10',
+            URL: '2018/',
+            banner: banner_2018
+        },
+        {
+            title: t('chongqing_web_frontend_conference_2018'),
+            date: '2018-08-18',
+            URL: 'https://fcc-cd.dev/activity/conference/web-fe-conf-chongqing-2018/',
+            banner: 'http://wimg.huodongxing.com/logo/201807/9450206156400/123133704540930_v2.jpg@!wmlogo'
+        },
+        {
+            title: t('chengdu_react_tech_conference_2018'),
+            date: '2018-03-31',
+            URL: 'https://juejin.cn/post/6844903694845100040',
+            banner: banner_2018_React
+        },
+        {
+            title: t('xinyao_cup_code_for_city_hackathon'),
+            date: '2018-01-20',
+            URL: '2018/Code4City/',
+            banner: gallery_Code4City[0]
+        },
+        {
+            title: t('chengdu_web_frontend_conference_2017'),
+            date: '2017-11-19',
+            URL: '2017/',
+            banner: banner_2017
+        }
+    ].sort(({ date: A }, { date: B }) => +new Date(B) - +new Date(A));
 
-const { date: first } = summits.slice(-1)[0],
-    { date: last } = summits[0];
+const { date: first } = summits().slice(-1)[0],
+    { date: last } = summits()[0];
 
 const years = Math.ceil(+last.slice(0, 4) - +first.slice(0, 4)) + 1;
 
-export const features = [
+export const features = () => [
     {
-        title: '秉持 freeCodeCamp 国际社区精神',
-        summary:
-            'freeCodeCamp.org 全球最大的开源编程学习平台（GitHub 星标前十），旨在成为计算机教育界的维基百科',
+        title: t('adhering_to_freecodecamp_international_community_spirit'),
+        summary: t('freecodecamp_global_largest_open_source_learning_platform'),
         logo: 'https://github.com/freeCodeCamp.png'
     },
     {
-        title: '引领 freeCodeCamp 中文社区潮流',
-        summary:
-            '第一个制定日常活动方针，第一个设计城市社区 logo、吉祥物，第一个举办黑客马拉松、技术大会，第一个试验开源线下运营',
+        title: t('leading_freecodecamp_chinese_community_trend'),
+        summary: t('first_to_establish_daily_activities_guidelines'),
         logo: 'https://github.com/freeCodeCamp-China.png'
     },
     {
-        title: '成都最活跃的程序员社区',
-        summary: `${years} 年连办 ${summits.length} 场技术干货大会，合办活动、中小分享会、动手训练营、线上直播不计其数`,
+        title: t('chengdu_most_active_programmer_community'),
+        summary: t('years_continue_to_undertake', {
+            which_year: years,
+            number_of_times: summits.length,
```

Comment:
```suggestion
            number_of_times: summits().length + '',
```
1. 函数不调用，怎么返回具体数据呢？
2. 里面定义了参数都是 `string` 类型，你却传了一个 `number`。

Reply:
第一个问题，我回答不出来，是我对TypeScript的函数调用不够深刻理解。第二个问题，之前我一直不知道，现在我才知道了summits().length是number。我按照上面的代码，结果还是出现同样的对话框。我试了tostring和number，还是不行，结果一样出现同样的对话框。

Reply:
> 第一个问题，我回答不出来，是我对TypeScript的函数调用不够深刻理解。

调用是一个运行时的概念，一切运行时概念都是 JavaScript 的，因为浏览器只能运行 JavaScript，而 TypeScript 只是开发和编译时的类型系统。

因此你要尽快多刷 fCC 基础题。

> 第二个问题，之前我一直不知道，现在我才知道了summits().length是number。我按照上面的代码，结果还是出现同样的对话框。我试了tostring和number，还是不行，结果一样出现同样的对话框。

你鼠标指到每一个传给参数的值，你会发现不止一个不是 `string`。

Reply:
好的，现在看见了。从明天开始，刷Leetcode的时间缩短，花多一些时间放在fCC基础题。